lib/cisco_node_utils/client/ssh/client.rb

```diff
@@ -0,0 +1,167 @@
+              else
+                munge_to_array('conf t')
+              end
+    values = munge_to_array(values)
+    super
+    req('cli_conf', context + values)
+  end
+
+  # Get the given state from the device.
+  #
+  # Unlike set() this will not clear the CLI cache;
+  # multiple calls with the same parameters may return cached data
+  # rather than querying the device repeatedly.
+  #
+  # @raise [Cisco::RequestNotSupported] if
+  #   structured output is requested but the given command can't provide it.
+  # @raise [Cisco::CliError] if the command is rejected by the device
+  #
+  # @param data_format one of Cisco::DATA_FORMATS. Default is :cli
+  # @param command [String] the show command to execute
+  # @param context [String, Array<String>] Context to refine the results
+  # @param value [String] Specific key to look up
+  # @param kwargs data-format-specific args
+  # @return [String, Hash]
+  def get(data_format: :cli,
+          command:     nil,
+          context:     nil,
+          value:       nil,
+          **_kwargs)
+    context = munge_to_array(context)
+    super
+    if data_format == :cli
+      output = req('cli_show_ascii', command)
+      return self.class.filter_cli(cli_output: output,
+                                   context:    context,
+                                   value:      value)
+    elsif data_format == :nxapi_structured
+      output = req('cli_show', command + ' | json')
+      output = JSON.parse(output)
+      return self.class.filter_data(data: output,
+                                    keys: context + munge_to_array(value))
+    else
+      fail TypeError
+    end
+  end
+
+  # Sends a request to the device and returns the output of the request or
+  # handles errors that happen.
+  # @raise Cisco::ClientError (should never occur)
+  # @raise Cisco::RequestNotSupported
+  # @raise Cisco::RequestFailed if any command is rejected as invalid
+  #
+  # @param type ["cli_show", "cli_show_ascii"] Specifies the type of command
+  #             to be executed.
+  # @param command_or_list [String, Array<String>] The command or array of
+  #                        commands which should be run.
+  # @return [Hash, Array<Hash>] output when type == "cli_show"
+  # @return [String, Array<String>] output when type == "cli_show_ascii"
+  def req(type, command_or_list)
+    if command_or_list.is_a?(Array)
+      command = command_or_list.join(' ; ')
+    else
+      command = command_or_list
+      command_or_list = [command]
+    end
+
+    debug("Input (#{type}): \'#{command}\'")
+    if cache_enable? && @cache_hash[type] && @cache_hash[type][command]
+      return @cache_hash[type][command]
+    end
+
+    # send the request and get the response
+    debug("Sending SSH request to device at #{@address}")
+    response = Net::SSH.start(@host, @username, password: @password, port: @port, timeout: @timeout, @verify_host_key => false) do |connection|
```

Comment:
What happens in the case where passwordless SSH is setup between the `puppet device` node and the switch?  Is the `@password` simply ignored in this context if set or `nil`?